app/src/main/java/com/imdb_compose/MainActivity.kt

```diff
@@ -1,47 +1,1195 @@
+                verticalArrangement = Arrangement.SpaceBetween
+            ) {
+                boxOfficeNumbers.forEachIndexed { i, movieMap ->
+                    Row (
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .heightIn(min = 64.dp, max = 80.dp),
+                        verticalAlignment = Alignment.CenterVertically
+                    ) {
+                        Text(
+                            modifier = Modifier
+                                .width(72.dp)
+                                .padding(start = 8.dp, end = 8.dp),
+                            text = "${ i + 1 }",
+                            fontSize = MaterialTheme.typography.displaySmall.fontSize,
+                            textAlign = TextAlign.Center
+                        )
+
+                        Box(
+                            modifier = Modifier
+                                .padding(end = 16.dp)
+                                .size(36.dp)
+                                .background(gray200.copy(alpha = 0.6f))
+                                .drawBehind {
+                                    val path = Path().apply {
+                                        moveTo(75f, 126f)
+                                        lineTo(0f, 166f)
+                                        lineTo(0f, 126f)
+                                        moveTo(45f, 126f)
+                                        lineTo(125f, 126f)
+                                        lineTo(125f, 166f)
+                                        close()
+                                    }
+                                    drawPath(path, color = gray200.copy(alpha = 0.6f))
+                                },
+                            contentAlignment = Alignment.Center
+                        ) {
+                            Icon(
+                                imageVector = Icons.Outlined.Add,
+                                contentDescription = "add favorite"
+                            )
+                        }
+
+                        Column (
+                            modifier = Modifier
+                        ) {
+                            Text(text = "${ movieMap["Release"] }")
+                            Text(text = "${ movieMap["Gross"] }", textAlign = TextAlign.Center)
+                        }
+
+                        Box(modifier = Modifier.fillMaxWidth(), contentAlignment = Alignment.CenterEnd) {
+                            Icon(
+                                modifier = Modifier.padding(end = 16.dp),
+                                imageVector = Icons.Outlined.BookOnline,
+                                contentDescription = "book online"
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun PosterBoxA(
+    catagory: String,
+    movie: MovieResult?,
```

Comment:
The example of where you're asking for too much info. Just ask for each individual piece of information separately as a its own parameter and don't have this function do the actual navigation. Have an `onClick` parameter and the caller will navigate for you so this function doesn't have think if you have a movie or TV show.